### ISSUE TITLE
TypeError: Cannot read property 'forEach' of undefined in Swagger

### ISSUE THREAD

**Issue by sinner** [Romakita/ts-express-decorators#440]
## Informations

Type | Version
---|---
Bug | 4.x

****

## Description
I've executed `npm start` after configuring a new project.

```sh
TypeError: Cannot read property 'forEach' of undefined
    at SwaggerService.getOpenAPISpec (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:165:35)
    at swagger.forEach (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:42:25)
    at Array.forEach (<anonymous>)
```

## Acceptance criterias

- [ ] Example (log, usecase, etc...)
- [x] Ts.ED  (bug/question)

```sh
[nodemon] restarting due to changes...
[nodemon] starting `ts-node src/index.ts`
Reading .env file from /var/srv/app/.env
Reading DB settings from file from /var/srv/app/ormconfig.json: 
 [object Object]
[2018-08-28T18:38:53.447] [INFO ] [TSED] - Scan files : /var/srv/app/controllers/current/**/*.ts
[2018-08-28T18:38:53.449] [INFO ] [TSED] - Scan files : /var/srv/app/controllers/v1/**/*.ts
[2018-08-28T18:38:53.449] [INFO ] [TSED] - Scan files : /var/srv/app/middlewares/**/*.ts
[2018-08-28T18:38:53.449] [INFO ] [TSED] - Scan files : /var/srv/app/services/**/*.ts
[2018-08-28T18:38:53.449] [INFO ] [TSED] - Scan files : /var/srv/app/converters/**/*.ts
[2018-08-28T18:38:53.452] [INFO ] [TSED] - settings.rootDir => /var/srv/app
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.env => development
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.httpPort => 3080
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.httpsPort => 3090
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.version => 1.0.0
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.uploadDir => /var/srv/app/uploads
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.debug => false
[2018-08-28T18:38:53.453] [INFO ] [TSED] - settings.logger => { logRequest: true }
[2018-08-28T18:38:53.454] [INFO ] [TSED] - settings.mount => { '/': '/var/srv/app/controllers/current/**/*.ts',
  '/v1': '/var/srv/app/controllers/v1/**/*.ts' }
[2018-08-28T18:38:53.454] [INFO ] [TSED] - settings.componentsScan => [ '/var/srv/app/middlewares/**/*.ts',
  '/var/srv/app/services/**/*.ts',
  '/var/srv/app/converters/**/*.ts' ]
[2018-08-28T18:38:53.455] [INFO ] [TSED] - settings.acceptMimes => [ 'application/json' ]
[2018-08-28T18:38:53.455] [INFO ] [TSED] - settings.httpsOptions => { key: <Buffer 2d 2d 2d 2d 2d 42 45 47 49 4e 20 50 52 49 56 41 54 45 20 4b 45 59 2d 2d 2d 2d 2d 0a 4d 49 49 45 76 51 49 42 41 44 41 4e 42 67 6b 71 68 6b 69 47 39 77 ... >,
  cert: <Buffer 2d 2d 2d 2d 2d 42 45 47 49 4e 20 43 45 52 54 49 46 49 43 41 54 45 2d 2d 2d 2d 2d 0a 4d 49 49 45 52 44 43 43 41 79 79 67 41 77 49 42 41 67 49 4a 41 50 ... >,
  passphrase: 'a6ff5a915c7d07c6f1b443d17af8d7c5ab3ed2d1',
  port: '3090' }
[2018-08-28T18:38:53.455] [INFO ] [TSED] - settings.typeorm => { db: 
   { name: 'default',
     type: 'postgres',
     host: 'project-db-postgresql',
     port: 5432,
     username: 'project',
     password: 'project',
     database: 'project',
     synchronize: true,
     logging: false,
     cache: { type: 'redis', options: [Object] },
     entities: [ 'src/entity/**/*.ts' ],
     migrations: [ 'src/entity/migrations/**/*.ts' ],
     subscribers: [ 'src/entity/subscribers/**/*.ts' ],
     cli: 
      { entitiesDir: 'src/entity',
        migrationsDir: 'src/entity/migrations',
        subscribersDir: 'src/entity/subscribers' } } }
[2018-08-28T18:38:53.455] [INFO ] [TSED] - settings.swagger => { path: '/api-docs' }
[2018-08-28T18:38:53.455] [INFO ] [TSED] - settings.customServiceOptions => {}
[2018-08-28T18:38:53.455] [INFO ] [TSED] - Build services
[2018-08-28T18:38:53.461] [INFO ] [TSED] - Connect to typeorm database: db
[2018-08-28T18:38:53.820] [INFO ] [TSED] - Build controllers
[2018-08-28T18:38:53.820] [INFO ] [TSED] - Routes mounted :
[2018-08-28T18:38:53.821] [INFO ] [TSED] - 
┌────────┬──────────┬──────────────┐
│ Method │ Endpoint │ Class method │
└────────┴──────────┴──────────────┘
TypeError: Cannot read property 'forEach' of undefined
    at SwaggerService.getOpenAPISpec (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:165:35)
    at swagger.forEach (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:42:25)
    at Array.forEach (<anonymous>)
    at SwaggerService.$afterRoutesInit (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:40:13)
    at ProviderRegistry_1.ProviderRegistry.forEach (/var/srv/app/node_modules/@tsed/src/common/di/services/InjectorService.ts:309:56)
    at Map.forEach (<anonymous>)
    at Registry.forEach (/var/srv/app/node_modules/@tsed/src/core/class/Registry.ts:140:19)
    at Function.<anonymous> (/var/srv/app/node_modules/@tsed/src/common/di/services/InjectorService.ts:300:26)
    at Generator.next (<anonymous>)
    at /var/srv/app/node_modules/tslib/tslib.js:107:75
(node:3190) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'forEach' of undefined
    at SwaggerService.getOpenAPISpec (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:165:35)
    at swagger.forEach (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:42:25)
    at Array.forEach (<anonymous>)
    at SwaggerService.$afterRoutesInit (/var/srv/app/node_modules/@tsed/src/swagger/services/SwaggerService.ts:40:13)
    at ProviderRegistry_1.ProviderRegistry.forEach (/var/srv/app/node_modules/@tsed/src/common/di/services/InjectorService.ts:309:56)
    at Map.forEach (<anonymous>)
    at Registry.forEach (/var/srv/app/node_modules/@tsed/src/core/class/Registry.ts:140:19)
    at Function.<anonymous> (/var/srv/app/node_modules/@tsed/src/common/di/services/InjectorService.ts:300:26)
    at Generator.next (<anonymous>)
    at /var/srv/app/node_modules/tslib/tslib.js:107:75
(node:3190) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:3190) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by Romakita** [comment]
Hi @sinner 
Have you create a controller ? Because this error mean you haven't controller resolved by the componentScan. I also agree, SwaggerService shouldn't fail.

See you
Romain

**Comment by Romakita** [comment]
Fixed with 4.30.2
